src/frontend/src/lib/components/convert/ConvertAmountSource.svelte

```diff
@@ -40,16 +39,37 @@
 	let isZeroBalance: boolean;
 	$: isZeroBalance = isNullish($sourceTokenBalance) || $sourceTokenBalance.isZero();
 
-	const setMax = () => {
-		if (!isZeroBalance) {
-			sendAmount = getMaxTransactionAmount({
+	let maxAmount: number | undefined;
+	$: maxAmount = nonNullish(totalFee)
+		? getMaxTransactionAmount({
 				balance: $sourceTokenBalance,
+				fee: BigNumber.from(totalFee),
 				tokenDecimals: $sourceToken.decimals,
 				tokenStandard: $sourceToken.standard
-			});
+			})
+		: undefined;
+
+	let amountSetToMax = false;
+	const setMax = () => {
+		if (!isZeroBalance && nonNullish(maxAmount)) {
+			amountSetToMax = true;
+
+			sendAmount = maxAmount;
 		}
 	};
 
+	/**
+	 * Reevaluate max amount if user has used the "Max" button and totalFee is changing.
+	 */
+	$: totalFee,
+		(() => {
+			if (!amountSetToMax) {
+				return;
+			}
+
+			debounce(() => setMax(), 500)();
```

Comment:
It's maybe better to define the debounce function outside of the auto subscriber to avoid it to be reevaluted and per extension not debouncing

```
const debounceSetMax = debounce(....

$: totalFee, ...
      debounceSetMax()
```

Reply:
Yep, it's done now.